### PR TITLE
Update, expand, and clean up Clustering sections

### PR DESCRIPTION
**Pull request by infoverload**
## Summary of the changes / Why this is an improvement

Part of the Q3 effort to [fix out-of-date content](https://github.com/crate/tech-writing-domain/issues/106)

- removed `scaling-intro.rst` since the content was quite redundant, inaccurate, didn't add much value, and the screenshots reveal that it was very old
- revamped "CrateDB multi-node setup" to remove a lot of redundancy, unused links, and re-organized some of the structure and phrasing
- revamped "CrateDB multi-zone setup" to add more explanations, examples, hyperlinks, pronoun consistency, and make the style more consistent with "CrateDB multi-node setup"

Tied to: https://github.com/crate/tech-writing-domain/issues/241, https://github.com/crate/crate-howtos/issues/228

## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: docs/clustering/multi-node-setup.rst [hunk 1]
```diff
@@ -27,152 +26,129 @@ method of your choice.
...
-options:
+   .. TIP::
 
-.. code-block:: console
+      Each directory will function as `CRATE_HOME`_ for that node
 
-    sh$ bin/crate \
-        -Cgateway.expected_nodes=3 \
-        -Cgateway.recover_after_nodes=3
+3. You should configure the :ref:`metadata gateway <metadata-gateway>` so that
```
Comment (joemoe) [inline docs/clustering/multi-node-setup.rst:110]: would it make sense to configure that before copying it? also i would be specific on where to do that.

File: docs/clustering/multi-node-setup.rst [hunk 2]
```diff
@@ -223,112 +203,68 @@ Alternatively, you can configure this at startup with a command-line option:
...
+^^^^^^^^^^^^^^^^^^^^
 
 The master node is responsible for making changes to the global cluster state.
-The cluster elects the master node from the configured list of master-eligible
-nodes during `master node election`_.
+The cluster `elects the master node`_ from the configured list of master-eligible
+nodes. 
 
-You can define the initial set of master-edible nodes with the
+You can define the initial set of master-eligible nodes with the
```
Comment (joemoe) [inline docs/clustering/multi-node-setup.rst:265]: I'd mention why in which case this would make sense.

**Comment by infoverload** [comment]
@joemoe 

Okay I totally forgot to push a bunch of changes (with more examples) I had made. I guess I was tired?

Could you review again? :p

File: docs/clustering/multi-node-setup.rst (hunk 1 above)
Comment (infoverload) [reply docs/clustering/multi-node-setup.rst:110]: Okay, I didn't actually try it out myself and was just rewording what was there.  But maybe I can try it out and discover for myself the flow. 

In terms of specifying where to do that, I was hoping that my linking to the example in the Metadata gateway section on the bottom would be sufficient.  Wdyt?

`The :ref:`metadata gateway <metadata-gateway>` section includes examples.`

File: docs/clustering/multi-node-setup.rst [hunk 3]
```diff
@@ -9,12 +9,11 @@ as a cluster of three or more nodes.
...
-Auto-bootstrapping requires zero configuration. However, in production
-environments, you must configure the bootstrapping process :ref:`manually
-<manual-bootstrapping>`.
+But in production environments, you must configure the bootstrapping process
+:ref:`manually <manual-bootstrapping>`.
 
-This guide shows you how to bootstrap a multi-node CrateDB cluster using the
-method of your choice.
+This guide shows you how to bootstrap (set up) a multi-node CrateDB cluster
+with different methods.
```
Comment (matthijskrul) [inline docs/clustering/multi-node-setup.rst:16]: I'd say "using" or "via"

File: docs/clustering/multi-node-setup.rst [hunk 4]
```diff
@@ -27,152 +26,129 @@ method of your choice.
...
 
 .. _auto-bootstrapping:
 
-Single host auto-bootstrapping
+Single-host auto-bootstrapping
 ------------------------------
 
-If you start up several nodes (with default configuration) on a single host,
-they will automatically discover one another and form a cluster.
+If you start up several nodes (with default configuration) on a single-host,
```
Comment (matthijskrul) [inline docs/clustering/multi-node-setup.rst:45]: No hyphen here

File: docs/clustering/multi-zone-setup.rst [hunk 5]
```diff
@@ -5,124 +5,188 @@ CrateDB multi-zone setup
 ========================
 
 If possible, we recommend running all CrateDB nodes of a cluster inside the
-same physical space (e.g. data center) to minimize network latency and maximize
+same physical space (i.e. data center) to minimize network latency and maximize
```
Comment (matthijskrul) [inline docs/clustering/multi-zone-setup.rst:8]: It doesn't have to be considered a 'data center', does it? You can run it locally, no?

File: docs/clustering/multi-zone-setup.rst [hunk 6]
```diff
@@ -5,124 +5,188 @@ CrateDB multi-zone setup
...
+For example, to force awareness on the pre-configured ``zone`` attribute for
+the ``us-east-1`` and ``us-west-1`` values, you can put the following in your
+`configuration`_ file:
 
 .. code-block:: yaml
 
     cluster.routing.allocation.awareness.force.zone.values: us-east-1,us-west-1
 
-When set, no more replicas than needed are allocated on a specific group of
+This means that no more `replicas`_ than needed are allocated on a specific group of
```
Comment (matthijskrul) [inline docs/clustering/multi-zone-setup.rst:165]: Dunno about 'on' here. Maybe 'for' or 'to'?

File: docs/clustering/index.rst [hunk 7]
```diff
@@ -4,8 +4,8 @@
 Clustering
 ==========
 
-One of the benefits of CrateDB is that you can scale out your cluster by adding
-more nodes. But what's the best way to do that?
+One of the benefits of CrateDB is that you can scale your cluster by adding
+more nodes.
```
Comment (mechanomi) [inline docs/clustering/index.rst:8]: I would keep "out". "out" here is being used in contrast to "up"

- "out" - scaling horizontally (one of CrateDB's selling points. something that you can do very easily using containers or VMS)
- "up" - scaling vertically (how most RDMBSes scale. and a complete PITA)

see https://stackoverflow.com/a/11715598

if you think we should clarify this in the text in some way, I'd be happy with that. but it's an important enough thing to highlight (in some way) in the intro to this section

thinking aloud...

perhaps this is a good candidate for an entry in the glossary (which we don't have yet). we could, perhaps, create the new glossary page under the appendices section, and add the following terms:

- scaling up (increasing capacity using vertical scaling -- link to "vertical scaling")
- scaling down (decreasing capacity using vertical scaling -- link to "vertical scaling")
- scaling out (increasing capacity using horizontal scaling -- link to "horizontal scaling")
- scaling in (decreasing capacity using horizontal scaling -- link to "horizontal scaling")
- horizontal scaling (full definition)
- vertical scaling (full definition)

that way, you could turn "scale out" in the text her into a link to the appropriate glossary term

File: docs/clustering/multi-zone-setup.rst (hunk 5 above)
Comment (mechanomi) [reply docs/clustering/multi-zone-setup.rst:8]: @matthijskrul agreed

also, minor: there should be a comma after the second period

File: docs/clustering/multi-zone-setup.rst [hunk 8]
```diff
@@ -5,124 +5,188 @@ CrateDB multi-zone setup
...
+=======================
 
-1. When inserting data, it should be replicated in a way so that at least one
-   full replica is present inside each zone.
-2. If a zone becomes unavailable, all data still needs to be fully available.
+For a multi-zone setup, CrateDB clusters need to fulfill the following:
+
+1. Data inserts should be replicated in a way where at least one full `replica`_
+   is present in each zone.
+2. All data still needs to be fully available if a zone becomes unavailable.
```
Comment (mechanomi) [inline docs/clustering/multi-zone-setup.rst:35]: please add an empty line between list items that are full sentences

(here and above)

File: docs/clustering/multi-zone-setup.rst [hunk 9]
```diff
@@ -5,124 +5,188 @@ CrateDB multi-zone setup
...
+As an example, you can create such a table by executing a statement like this
+in the `CrateDB Shell`_:
+
+.. code-block:: sql
+
+    cr> create table my_table (
+          first_column integer,
+          second_column text
+        ) clustered into 6 shards
+        with (number_of_replicas = 1);
```
Comment (mechanomi) [inline docs/clustering/multi-zone-setup.rst:123]: SQL reserved keywords should be capitalized in example blocks and literals

https://www.postgresql.org/docs/8.1/sql-keywords-appendix.html

File: docs/clustering/multi-zone-setup.rst (hunk 6 above)
Comment (mechanomi) [reply docs/clustering/multi-zone-setup.rst:165]: I think "on" is correct here. the allocated replicas are not allocated *to* the node. they are allocated *as* replicas. the terminology is a little confusing. but this is how I understand it

@kovrus perhaps you can help clear this up? 😅

File: docs/clustering/multi-zone-setup.rst [hunk 10]
```diff
@@ -5,124 +5,188 @@ CrateDB multi-zone setup
...
-requirements above, but be aware of the drawbacks such a setup can have.
+   If you have 2 nodes with the ``zone`` attribute set to ``us-east-1`` and you
+   `create a table`_ with 8 shards and 1 replica, 8 primary shards will be allocated
+   and the 8 replica shards will be left unassigned. Only when you add a new node
+   with the ``zone`` attribute set to ``us-west-1`` will the replica shards be
+   allocated.
+
+By using these settings correctly and understanding the concepts behind them,
+you should be able to set up a functioning cluster that spans across multiple
+zones. However, be aware of the drawbacks that a multi-zone setup can have.
```
Comment (mechanomi) [inline docs/clustering/multi-zone-setup.rst:178]: can we mention some of the drawbacks, or link to a document that does?

File: docs/clustering/multi-zone-setup.rst (hunk 9 above)
Comment (infoverload) [reply docs/clustering/multi-zone-setup.rst:123]: Yikes! This will have to be fixed in several parts of the docs as well.

**Pull request merged**